cdc/sink/dmlsink/mq/kafka_dml_sink.go

```diff
@@ -107,22 +108,38 @@ func NewKafkaDMLSink(
 		return nil, cerror.WrapError(cerror.ErrKafkaInvalidConfig, err)
 	}
 
+	var (
+		claimCheck        *ClaimCheck
+		claimCheckEncoder codec.ClaimCheckEncoder
+		ok                bool
+	)
+
+	if encoderConfig.LargeMessageHandle.EnableClaimCheck() {
+		claimCheckEncoder, ok = encoderBuilder.Build().(codec.ClaimCheckEncoder)
+		if !ok {
+			return nil, cerror.ErrKafkaInvalidConfig.
+				GenWithStack("claim-check enabled but the encoding protocol %s does not support", protocol.String())
+		}
+
+		storageURI := replicaConfig.Sink.KafkaConfig.LargeMessageHandle.ClaimCheckStorageURI
+		claimCheck, err = NewClaimCheck(ctx, storageURI, changefeedID)
```

Comment:
whether there are errors  without the right permission to access the storage url?

Reply:
it's possible. the permission is configurable in the storage uri.

Reply:
you can refer to the approach used in the cloud object storage  sink which can detect the error early .